lxd/storage_zfs.go

```diff
@@ -1842,75 +1842,73 @@ func (s *storageZfs) zfsPoolCreate() error {
 
 	// Create default dummy datasets to avoid zfs races during container
 	// creation.
-	err := s.zfsPoolVolumeCreate("containers")
-	if err != nil {
-		return err
-	}
-
-	err = s.zfsPoolVolumeSet("containers", "mountpoint", "none")
+	poolName := s.getOnDiskPoolName()
+	dataset := fmt.Sprintf("%s/containers", poolName)
+	fixperms := shared.VarPath("storage-pools", s.pool.Name, "containers")
+	msg, err := zfsPoolVolumeCreate(dataset, fmt.Sprintf("mountpoint=%s", fixperms))
```

Comment:
NP, totally fair point. I just pointlessly replicated the same mistake I made when writing the original code in a more concise manner. :) I'll send an updated branch and also make sure that mountpoints of existing pools and datasets will be cleared as well.